### PR TITLE
Adding SkipInit to Internal.Runtime.CompilerServices.Unsafe

### PR DESCRIPTION
**Pull request by tannergooding**
This exposes `Internal.Runtime.CompilerServices.Unsafe.SkipInit` and consumes it in `Decimal.DecCalc`. There were no pmi diffs from doing so.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @jkotas.

Are there other known places that could switch to using `Unsafe.SkipInit`? From a quick search, there didn't seem to be any common pattern that could be used to easily identify said locations.

File: src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs [hunk 1]
```diff
@@ -990,8 +990,7 @@ internal static unsafe void DecAddSub(ref DecCalc d1, ref DecCalc d2, bool sign)
 
                     // Have to scale by a bunch. Move the number to a buffer where it has room to grow as it's scaled.
                     //
-                    Buf24 bufNum;
-                    _ = &bufNum; // workaround for CS0165
+                    Unsafe.SkipInit(out Buf24 bufNum); // workaround for CS0165
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs]: I do not think we need the " workaround for CS0165" comments anymore. `Unsafe.SkipInit` is self-documenting.

**Comment by jkotas** [comment]
> Are there other known places that could switch to using Unsafe.SkipInit?

You have fixed all the ones that I am aware of.

File: src/vm/jitinterface.cpp [hunk 2]
```diff
@@ -7338,6 +7338,17 @@ bool getILIntrinsicImplementationForUnsafe(MethodDesc * ftn,
         methInfo->options = (CorInfoOptions)0;
         return true;
     }
+    else if (tk == MscorlibBinder::GetMethod(METHOD__UNSAFE__SKIPINIT)->GetMemberDef())
```
Comment (jkotas) [inline src/vm/jitinterface.cpp]: Could you please add this to src\tools\crossgen2\Common\TypeSystem\IL\Stubs\UnsafeIntrinsics.cs as well?

Comment (jkotas) [reply src/vm/jitinterface.cpp]: Missed it. I see that it is there already.

File: src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs (hunk 1 above)
Comment (tannergooding) [reply src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs]: Removed comments in the latest commit.

**Pull request merged**